analyzer/src/main/kotlin/PackageManager.kt

```diff
@@ -149,18 +149,13 @@ abstract class PackageManager(
          *                     first element of the list that is recognized as a VCS URL is used.
          */
         fun processPackageVcs(vcsFromPackage: VcsInfo, fallbackUrls: List<String> = emptyList()): VcsInfo {
-            val normalizedVcsFromPackage = vcsFromPackage.normalize()
-
-            val normalizedUrl = normalizedVcsFromPackage.url.takeIf { it.isNotEmpty() }
-                ?: fallbackUrls
-                    .asSequence()
-                    .map { normalizeVcsUrl(it) }
-                    .filterNot { VersionControlSystem.forUrl(it) == null }
-                    .firstOrNull()
-                    .orEmpty()
-
-            val vcsFromUrl = VcsHost.toVcsInfo(normalizedUrl)
-            return vcsFromUrl.merge(normalizedVcsFromPackage)
+            val applicableVcsInfo = sequenceOf(vcsFromPackage.url, *fallbackUrls.toTypedArray()).map {
```

Comment:
This syntax would be a bit shorter: `(sequenceOf(vcsFromPackage.url) + fallbackUrls).map { ...`

Reply:
Indeed, and that's how I had it before, but then I decided it would be better to not create a temporary sequence.

Reply:
On the other hand, `toTypedArray()` now creates a temporary array. So should we maybe change `fallbackUrls` to be an array instead of a list?

Reply:
But this code creates a temporary array instead. I was curious if there is a performance difference between the two solutions, so I wrote a small test and it turns out depending on which elements of the sequence are accessed afterwards using `+` is about 20-70% faster, I assume because it's wrapping the iterator of the list instead of copying all references into an array. But because this functions on handles small lists the performance difference is of course negligible.

Reply:
> But this code creates a temporary array instead.

I guess that refers to the code *before* I roce-pushed a new version that uses `vararg`.

What I like about the `vararg` approach is that in the majority of cases you can just write `homepageUrl` instead of `listOf(homepageUrl)`.